docs/0002-fixed-length-n-tuple-trees.md

```diff
@@ -0,0 +1,223 @@
+
+The nature of the N-tuple algorithm means that the following additional conditions on parameters MUST be observed:
+* **numberOfTuples * tuplesSize <= identifierLength**
+* If **tupleSize = 0** then **numberOfTuples = 0**
+
+### shortObjectRoot
+
+Once a Storage Hierarchy has been created, the name of each OCFL Object Root directory should also be determined from the identifier in a consistent manner. The default approach is to use the full (stripped) identifier but, if identifiers are long or there is the need to keep Storage Hierarchy paths short because of object complexity, there is the option to just use the portion of the identifier that remains after Storage Hierarchy path generation. To continue the UUID example the full OCFL Object Root path could be:    
+* **shortObjectRoot = false** /f81/d4f/ae7/f81d4fae7dec11d0a76500a0c91e6bf6/
+* **shortObjectRoot = true** /f81/d4f/ae7/dec11d0a76500a0c91e6bf6/
+
+Note, if **numberOfTuples * tuplesSize = identifierLength** there is no remaining portion of the identifer and so
+**shortObjectRoot** MUST be false.
+
+## Examples
+
+These examples are taken from the OCFL Implementation notes:
+
+* *Flat*: Each object is contained in a directory with a name that is simply derived from the unique identifier of the object.
+  * identiferLength = 12
+  * caseMapping = "toLower"
+  * invertMapping = false
+  * tupleSize = 0
+  * numberOfTuples = 0
+  * shortObjectRoot = false
+
+                [storage_root]
+                    ├── 0=ocfl_1.0
+                    ├── ocfl_1.0.html (optional copy of the OCFL specification)
+                    ├── d45be626e024
+                    |   ├── 0=ocfl_object_1.0
+                    |   ├── inventory.json
+                    |   ├── inventory.json.sha512
+                    |   └── v1...
+                    ├── d45be626e036
+                    |   ├── 0=ocfl_object_1.0
+                    |   ├── inventory.json
+                    |   ├── inventory.json.sha512
+                    |   └── v1...
+                    ├── 3104edf0363a
+                    |   ├── 0=ocfl_object_1.0
+                    |   ├── inventory.json
+                    |   ├── inventory.json.sha512
+                    |   └── v1...
+                    └── ...
+                
+
+* PairTree: [PairTree] is designed to overcome the limitations on the number of files in a directory that most file systems have. It creates hierarchy of directories by mapping identifier strings to directory paths two characters at a time. For numerical identifiers specified in hexadecimal this means that there are a maximum of 256 items in any directory which is well within the capacity of any modern filesystem. However, for long identifiers, pairtree creates a large number of directories which will be sparsely populated unless the number of objects is very large. Traversing all these directories during validation or rebuilding operations can be slow.
```

Comment:
typo: "It creates hierarchy [a] of directories..."

Reply:
Fixed :)